### DISCUSSION TITLE
我希望在电视剧重命名时，使用类似 s04/S04E01 - title.mkv 的形式，但尝试了几种方案都不行

### DISCUSSION THREAD

**Discussion by skiinder** [jxxghp/MoviePilot#2364] [问题咨询] [answered]
我目前使用的表达式：
{{title}} ({{year}})/s{{"%02d"|format(season)}}/{{season_episode}} - {{episode_title}}{{fileExt}}
但是由于season变量是str类型失败；

尝试加入类型强转
{{title}} ({{year}})/s{{"%02d"|format(int(season))}}/{{season_episode}} - {{episode_title}}{{fileExt}}
会报int未定义；

尝试
{{title}} ({{year}})/s{{"%02s"|format(int(season))}}/{{season_episode}} - {{episode_title}}{{fileExt}}
会用空格补位

以前没用过jinja这个语言，这种情况有没有解决方案？

**Comment by skiinder** [comment, marked as answer]
我找到解决方案了，尽管有点笨，但是可以用：
{{title}} ({{year}})/s{% if season|length == 1 %}0{{ season }}{% else %}{{ season }}{% endif %}/{{season_episode}} - {{episode_title}}{{fileExt}}

**Comment by AisukaYuki** [comment]
`S{{ "%02d" | format(season | int) }}`

你一开始的想法其实是ok的，只是写错了。

**Reply by skiinder** [reply]
原来语法是这样的，jinja文档里面没有啊。我觉得movie pilot可以增加一些常用模板普及一下语法

**Reply by 777Arch** [reply]
同样的需求，学习到了
